### PR TITLE
Replace checkpoints with sequence ID

### PR DESCRIPTION
**Pull request by Geod24**
```
Next checkpoint do not play a role in the consensus. They take up space and are solely debugging information.
Their original role was to provide a stronger rule w.r.t balance verification,
however such rules proved to be impossible to implement without compromising
other aspects of the design (namely, it made issuing transactions very complicated).
```

### CODE REVIEW

**Comment by Geod24** [comment]
First step towards fixing https://github.com/bosnet/sebak/issues/123

File: lib/operation_payment.go [hunk 1]
```diff
@@ -65,7 +65,7 @@ func FinishOperationPayment(st *sebakstorage.LevelDBBackend, tx Transaction, op
 		return
 	}
 
-	if err = baTarget.Deposit(op.B.GetAmount(), tx.NextTargetCheckpoint()); err != nil {
+	if err = baTarget.Deposit(op.B.GetAmount()); err != nil {
```
Comment (Geod24) [inline lib/operation_payment.go]: This was a bug BTW

File: lib/transaction_checker.go [hunk 2]
```diff
@@ -28,12 +28,7 @@ func CheckTransactionSource(c sebakcommon.Checker, args ...interface{}) (err err
 }
 
 func CheckTransactionCheckpoint(c sebakcommon.Checker, args ...interface{}) (err error) {
-	checker := c.(*TransactionChecker)
-	if _, err = sebakcommon.ParseCheckpoint(checker.Transaction.B.Checkpoint); err != nil {
-		err = sebakerror.ErrorTransactionInvalidCheckpoint
-		return
-	}
-
+	//checker := c.(*TransactionChecker)
```
Comment (Geod24) [inline lib/transaction_checker.go]: I'm not sure if that's correct. Of course `ParseCheckpoint` needs to be removed, but it feels like this is missing a check ?

**Comment by Geod24** [comment]
Rebased on master

**Comment by Geod24** [comment]
Ping

**Comment by spikeekips** [comment]
@Geod24 #123 is still in proposal, so I think this code and further development would be better in new branch.

**Comment by Geod24** [comment]
@kfangw @spikeekips : Following our discussions, I rebased and completed this.
It now completely replaces checkpoints with sequence IDs.

File: lib/api/resource.go [hunk 3]
```diff
@@ -48,15 +48,15 @@ const (
...
+	sequenceID uint64
 	balance    string
 }
 
 func (aa APIResourceAccount) GetMap() hal.Entry {
 	return hal.Entry{
 		"id":         aa.accountId,
 		"account_id": aa.accountId,
-		"checkpoint": aa.checkpoint,
+		"sequenceid": aa.sequenceID,
```
Comment (kfangw) [inline lib/api/resource.go]: `sequence_id`

File: lib/api/resource.go [hunk 4]
```diff
@@ -77,28 +77,25 @@ func (aa APIResourceAccount) Serialize() (encoded []byte, err error) {
...
-		"fee_paid":          at.fee,
-		"source_checkpoint": at.sourceCheckpoint,
-		"target_checkpoint": at.targetCheckpoint,
-		"created_at":        at.created,
-		"operation_count":   len(at.operations),
+		"id":              at.hash,
+		"hash":            at.hash,
+		"account":         at.source,
+		"fee_paid":        at.fee,
+		"sequencid":       at.sequenceID,
```
Comment (kfangw) [inline lib/api/resource.go]: ```sequence_id```

**Comment by spikeekips** [comment]
#204

**Pull request merged**